### ISSUE TITLE
Bug: [no-extra-parens] False positive when calling generic function with type tuple parameter containing parenthesis

### ISSUE THREAD

**Issue by burtek** [typescript-eslint/typescript-eslint#7030]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=CYUwxgNghgTiAEYD2A7AzgF3hkmBc8A3gFDzwhRgAWBJZZAPAIIwDma5AHjisBwK4oA1iiQB3FAG0AugD4AFLBgEW7GQEoC8jAEsMEEAUwwdKVgBp4AMxRaAdA9jsVbNOvgBeWfABuSHcDuXr7+wKTwAL7EUcQ4mHYU1AySTgCMWij8ALYARiAw8AA+8MamrOoylk4ATASZufly8pLhkpKp0pbVna2SAOTVfZ3wAMzSxNLq8n26+iB9lopsqVVs1UHedPDR6sRAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDWsX7RETVBkiJo0DtEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
declare const test: {
  each: {
    <Args extends unknown[]>(arr: Args[]): (title: string, fn: (...args: Args) => void) => void
  }
}

test.each<[arg1: (number | string)[], arg2: number]>([
  [[1], 2],
  [['2'], 3]
])('title', (arg1, arg2) => {
  
})
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-extra-parens": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error

### Actual Result

Unnecessary parentheses around expression. 7:53 - 7:54

### Additional Info

Similar to #5488, but more sophisticated issue found while writing tests with jest.

**Comment by xsjcTony** [comment]
Isn't this the same issue? https://github.com/typescript-eslint/typescript-eslint/issues/5488

**Comment by burtek** [comment]
@xsjcTony The example from #5488 works correctly with listed versions, but mine does not. Might be the same issue but that would imply #5488 wasn't fixed correctly. Anyway, #5488 was marked as fixed ages ago and the issue is there, hence new ticket.

**Comment by oxyu8** [comment]
I'll do this.
